src/editor.c

```diff
@@ -1906,8 +1915,25 @@ static gchar *find_calltip(const gchar *word, GeanyFiletype *ft)
 
 	g_return_val_if_fail(ft && word && *word, NULL);
 
-	/* use all types in case language uses wrong tag type e.g. python "members" instead of "methods" */
-	tags = tm_workspace_find(word, NULL, tm_tag_max_t, NULL, ft->lang);
+	tags = get_scoped_tags(editor, pos - strlen(word));
+	if (tags)
+	{
+		GPtrArray *filtered = g_ptr_array_new();
+
+		foreach_ptr_array(tag, i, tags)
+		{
+			if (strcmp(tag->name, word) == 0)
+				g_ptr_array_add(filtered, tag);
+		}
+
+		g_ptr_array_free(tags, TRUE);
+		tags = filtered;
+	}
+	else
+	{
+		/* use all types in case language uses wrong tag type e.g. python "members" instead of "methods" */
```

Comment:
In symbols.c there's yet another table which defines section names and icons used for individual tags:

https://github.com/geany/geany/blob/master/src/symbols.c#L463